src/core/load_balancing/rls/rls.cc

```diff
@@ -2090,14 +2147,20 @@ void RlsLb::ResetBackoffLocked() {
 void RlsLb::ShutdownLocked() {
   GRPC_TRACE_LOG(rls_lb, INFO) << "[rlslb " << this << "] policy shutdown";
   registered_metric_callback_.reset();
-  MutexLock lock(&mu_);
-  is_shutdown_ = true;
-  config_.reset(DEBUG_LOCATION, "ShutdownLocked");
-  channel_args_ = ChannelArgs();
-  cache_.Shutdown();
-  request_map_.clear();
-  rls_channel_.reset();
-  default_child_policy_.reset();
+  RefCountedPtr<ChildPolicyWrapper> child_policy_to_delete;
+  std::vector<RefCountedPtr<ChildPolicyWrapper>> child_policies_to_delete;
+  OrphanablePtr<RlsChannel> rls_channel_to_delete;
+  ChannelArgs channel_args_to_delete;
```

Comment:
I don't think `channel_args_` is guarded by the mutex.  We can just clear it after we release the lock.

Reply:
is it protected by WorkSerializer instead?

Reply:
Yes.